File: exercism/kotlin/hamming/src/main/kotlin/Hamming.kt
object Hamming {
  fun compute(s1: String, s2: String): Int {
    if (s1.length != s2.length) {
      throw IllegalArgumentException("Arguments must be of the same length.")
    }

    return s1.indices.count { s1[it] != s2[it] }
  }
}

File: exercism/elixir/hamming/hamming.exs
defmodule Hamming do
  @spec hamming_distance([char], [char]) :: any
  def hamming_distance(strand1, strand2) do
    if length(strand1) == length(strand2) do
      {:ok,
       strand1 |> Enum.zip(strand2)
               |> Enum.count(fn ({x, y}) -> x != y end)
      }
    else
      {:error, "Lists must be the same length"}
    end
  end
end
